src/main/java/com/searchservice/app/rest/ManageTableResource.java

```diff
@@ -122,17 +128,37 @@ public ManageTableResource(ManageTableServicePort manageTableServicePort) {
 			@PathVariable String tableName, 
 			@PathVariable int clientid) {
 		log.debug("Delete table");
-
 		tableName = tableName + "_" + clientid;
-		ResponseDTO apiResponseDTO = manageTableServicePort.deleteTable(tableName);
-
-		if (apiResponseDTO.getResponseStatusCode() == 200) {
+		if(tableDeleteServicePort.checkTableExistensce(tableName)) {
+		    ResponseDTO apiResponseDTO = tableDeleteServicePort.initializeTableDelete(clientid, tableName);
+		    if (apiResponseDTO.getResponseStatusCode() == 200) {
+			 return ResponseEntity.status(HttpStatus.OK).body(apiResponseDTO);
+		   } else {
+			 log.debug("Exception occurred: {}", apiResponseDTO);
+			 throw new BadRequestOccurredException(400, BAD_REQUEST_MSG);
+		   }
+		}else {
+			throw new BadRequestOccurredException(400, "Table "+tableName+" Does Not Exist");
```

Comment:
To be accurate, we can put Table 'xyz' for clientId '123' Does Not Exist

Reply:
yes surely should i make the changes

Reply:
yes